packages/starlight/utils/plugins.ts

```diff
@@ -72,6 +73,13 @@ export async function runPlugins(
 		});
 	}
 
+	if (context.config.output === 'static' && !starlightConfig.prerender) {
+		throw new AstroError(
+			'Starlight prerendering cannot be disabled when the project is set to "static" output mode.',
+			'Either set output mode to "server"/"hybrid" or enable Starlight prerendering.'
```

Comment:
Suggesting a small rework here. My thinking here is that usually people don’t actively set `output: 'static'` — it’s the implicit default — so they may not understand how/why their project is set that way. Also seems a good opportunity to add a docs link.

```suggestion
			'Starlight’s `prerender: false` option requires `output: "hybrid"` or `"server"` in your Astro config.',
			'Either set `output` in your Astro config or set `prerender: true` in the Starlight options.\n\n' +
				'Learn more about rendering modes in the Astro docs: https://docs.astro.build/en/basics/rendering-modes/'
```

(Instructing the user to set `prerender: true` may feel a bit odd, given it’s the default, but I think in this scenario it gives them a really clear course of action — “I already have `prerender: false`, I just need to updated that” — vs trying to explain removing the option entirely, and an error message’s aim is getting things working again as simply as possible.)